#!/usr/bin/env python
PACKAGE = "hkvs"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("heightDiff", double_t, 0, "heightDiff", 0,0,5000)
gen.add("widthDiff", double_t, 0, "widthDiff",    0,0,5000)
gen.add("angleDiff", double_t, 0, "angleDiff",   0,0,5000)
gen.add("meanheight", double_t, 0, "meanheight",    0,0,5000)
gen.add("yDiffRatio", double_t, 0, "yDiffRatio",    0,0,5000)
gen.add("xDiffRatio", double_t, 0, "xDiffRatio",    0,0,5000)
gen.add("dis", double_t, 0, "dis",   0,-10000,10000)
gen.add("ratio", double_t, 0, "ratio",    0,-10000,10000)
gen.add("heightDiff_ratio",double_t,0,"heightDiff_ratio",0,-10000,10000)



gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)


size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                      gen.const("Medium",     int_t, 1, "A medium constant"),
                      gen.const("Large",      int_t, 2, "A large constant"),
                      gen.const("ExtraLarge", int_t, 3, "An extra large constant")], "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "hkvs", "hkvs"))
